packages/headless/doc-parser/src/extractor.ts

```diff
@@ -25,58 +26,83 @@ enum Level {
   Type,
 }
 
-export function extractTypes(headingEntities: AnyEntity[]): Extraction {
+export function extractTypes(
+  headingEntities: AnyEntity[],
+  referencesCount: InterfaceReferencesCount
+): Extraction {
   const extraction: Extraction = {
     types: [],
   };
 
-  processObjectEntities(headingEntities, extraction, Level.Heading);
-  processFunctionEntities(headingEntities, extraction);
+  processObjectEntities(
+    headingEntities,
+    extraction,
+    Level.Heading,
+    referencesCount
+  );
+  processFunctionEntities(headingEntities, extraction, referencesCount);
 
   return extraction;
 }
 
 function processObjectEntities(
   entities: AnyEntity[],
   extraction: Extraction,
-  level: Level
+  level: Level,
+  referencesCount: InterfaceReferencesCount
 ) {
   entities
     .filter(isObjectEntity)
-    .forEach((entity) => extract(entity, extraction, level + 1));
+    .forEach((entity) =>
+      extract(entity, extraction, level + 1, referencesCount)
+    );
 }
 
 function processFunctionEntities(
   entities: AnyEntity[],
-  extraction: Extraction
+  extraction: Extraction,
+  referencesCount: InterfaceReferencesCount
 ) {
   entities.filter(isFunctionEntity).forEach((func) => {
     const {params, returnType} = func;
     const entities = isString(returnType) ? params : [...params, returnType];
 
-    processObjectEntities(entities, extraction, Level.Key);
+    processObjectEntities(entities, extraction, Level.Key, referencesCount);
   });
 }
 
-function extract(entity: ObjEntity, extraction: Extraction, level: Level) {
+function extract(
+  entity: ObjEntity,
+  extraction: Extraction,
+  level: Level,
+  referencesCount: InterfaceReferencesCount
+) {
   const {members} = entity;
 
   if (!members.length) {
```

Comment:
Currently, because of this condition, the check of `isTypeExtracted` is redundant. I chose to keep `!entity.isTypeExtracted` below since it felt a lot more explicit and readable.